liblwgeom/lwgeom.c

```diff
@@ -2623,22 +2621,17 @@ lwgeom_boundary(LWGEOM *lwgeom)
 	case POLYGONTYPE: {
 		LWPOLY *lwpoly = (LWPOLY *)lwgeom;
 
-		if (lwpoly->nrings == 1)
+		LWMLINE *lwmline = lwmline_construct_empty(srid, hasz, hasm);
+		for (uint32_t i = 0; i < lwpoly->nrings; i++)
 		{
-			POINTARRAY *ring = ptarray_clone_deep(lwpoly->rings[0]);
-			return (LWGEOM *)lwline_construct(srid, 0, ring);
+			POINTARRAY *ring = ptarray_clone_deep(lwpoly->rings[i]);
+			lwmline_add_lwline(lwmline, lwline_construct(srid, 0, ring));
 		}
-		else
-		{
-			LWMLINE *lwmline = lwmline_construct_empty(srid, hasz, hasm);
-			for (uint32_t i = 0; i < lwpoly->nrings; i++)
-			{
-				POINTARRAY *ring = ptarray_clone_deep(lwpoly->rings[i]);
-				lwmline_add_lwline(lwmline, lwline_construct(srid, 0, ring));
-			}
 
-			return (LWGEOM *)lwmline;
-		}
+		LWGEOM *lwout = lwgeom_homogenize((LWGEOM *)lwmline);
+		lwgeom_free((LWGEOM *)lwmline);
```

Comment:
Is homogenize really needed here ? We don't expect to be able to construct non-homogenous things via lwmline_add_lwline, are we ? Don't we get an exception if we tried ?

Reply:
I've checked - it is converting single-ring polygons to non-multi linestring. Added a comment on merge.